### PR TITLE
Add `list_repo_commits` to list git history of a repo

### PR DESCRIPTION
**Pull request by Wauplin**
FIX https://github.com/huggingface/huggingface_hub/issues/1288 (following [server-side implementation](https://github.com/huggingface/moon-landing/pull/5301) -internal link- thanks @Kakulukian).

This PR adds a new helper `list_repo_commits` (similar to `list_repo_refs` and list_repo_files`). It lists the all history of a repository on the Hub for a given revision. Commits are sorted by date (last commit first).

@camenduru this will finally solve your problem on "how to create an empty branch?". I did not make a special parameter in `create_branch` that would use `list_repo_commits` under the hood as I thought it was a quite specific use case.

```py
>>> from huggingface_hub import HfApi
>>> api = HfApi()

# Get initial commit on a repo
>>> initial_commit = api.list_repo_commits("gpt2")[-1]

# Initial commit is always a system commit containing the `.gitattributes` file.
>>> initial_commit
GitCommitInfo(
    commit_id='9b865efde13a30c13e0a33e536cf3e4a5a9d71d8',
    authors=['system'],
    created_at=datetime.datetime(2019, 2, 18, 10, 36, 15, tzinfo=datetime.timezone.utc),
    title='initial commit',
    message='',
    formatted_title=None,
    formatted_message=None
)

# Create an "empty branch" by deriving from initial commit (branch is not entirely empty, `.gitattributes` is still there :) )
>>> api.create_branch("gpt2", "new_empty_branch", revision=initial_commit.commit_id)
```

### CODE REVIEW

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -705,6 +706,49 @@ class GitRefs:
...
+            List of authors of the commit.
+        created_at (`datetime`):
+            Datetime when the commit was created.
+        title (`str`):
+            Title of the commit. This is a free-text value entered by the authors.
+        message (`str`):
+            Description of the commit. This is a free-text value entered by the authors.
+        formatted_title (`str`):
+            Title of the commit formatted as HTML. Only returned if `formatted=True` is set.
+        message (`str`):
```
Comment (julien-c) [inline src/huggingface_hub/hf_api.py:727]:
```suggestion
        formatted_message (`str`):
```

**Review by julien-c — COMMENTED** [review]
really cool!

**Comment by camenduru** [comment]
🎉 thanks @Kakulukian ♥️ thanks @Wauplin ♥️

**Review by LysandreJik — APPROVED** [review]
Thank you @Wauplin, very clean!

**Comment by camenduru** [comment]
🥳

**Pull request merged**